src/python_testing/TC_MOD_1_2.py

```diff
@@ -0,0 +1,180 @@
+
+        if not hasattr(supported_mode, 'mode') and not isinstance(supported_mode.mode, int):
+            asserts.fail("Supported modes struct does not have attribute mode or does not contain int")
+
+        if not hasattr(supported_mode, 'semanticTags'):
+            asserts.fail("Supported mode struct does not have attribute semanticTags")
+
+        # veirfy if we have entries for semanticTags if there are any  verfy values no larger that 16 bits
+        # entries must have the value and manufacturer code.
+        self._log_attribute("Semantic tags", supported_mode.semanticTags)
+
+        if isinstance(supported_mode.semanticTags, list) and len(supported_mode.semanticTags) > 0:
+            logger.info(
+                "SupportedMode.semanticTags contains values, verifying attributes for manufacturedcode and value are not longer than 16bits int")
+            for semantictag in supported_mode.semanticTags:
+                asserts.assert_true(semantictag.mfgCode >= 0 and semantictag.mfgCode <= self._16bitshex,
+                                    "Element semantictag.Mfgcode is greater than 16 bits")
+                asserts.assert_true(semantictag.value >= 0 and semantictag.value <= self._16bitshex,
+                                    "Element semantictag.Value is greater than 16 bits")
+
+    def _log_attribute(self, name, value):
+        logger.info(f"{name} attribute with value: {value} with type: {type(value)}")
+
+    @async_test_body
+    async def test_MOD_1_2(self):
+        self.cluster = Clusters.ModeSelect
+        self.endpoint = self.get_endpoint(1)
+        self._16bitshex = 0xFFFF
+
+        # Commision device
+        # In the test plan step 1 is defined as a precondition.
+        self.step(1)
+
+        # Veirfy the Supported modes
+        # Verify contains attributes label and mode
+        # Verify if semantic tags has elements in list , and if there are values assert the values
+        self.step(2)
+        supported_modes = await self.read_single_attribute_check_success(endpoint=self.endpoint, cluster=self.cluster, attribute=self.cluster.Attributes.SupportedModes)
+        logger.info(f"Supported modes {supported_modes}")
+        # List must not be empty
+        asserts.assert_true(len(supported_modes) > 0, "Supported modes can not be empty.")
+        supported_modes_values = []
+        for supported_mode in supported_modes:
+            logger.info(
+                f"Label {supported_mode} with type {type(supported_mode)} and {supported_mode.label} and {supported_mode.mode} and {supported_mode.semanticTags}")
+            # Verify the struct values
+            self._verify_supported_mode(supported_mode=supported_mode)
+            # After verifying the struct is correct append the mode value.
+            supported_modes_values.append(supported_mode.mode)
+        logger.info(f"Supported modes values {supported_modes_values}")
```

Comment:
Can you add a check in here and in the test plan that the mode numbers are unique?
asserts.assert_equal(len(supported_modes_values), len(set(supported_modes_values)), "duplicate values found for supported modes")